lib/code_corps/stripe_service/events/account_updated.ex

```diff
@@ -5,144 +5,21 @@ defmodule CodeCorps.StripeService.Events.AccountUpdated do
 
   @api Application.get_env(:code_corps, :stripe)
 
-  # def handle(%{"data" => %{"object" => %{"livemode" => false}}}), do: {:ok, :ignored_not_live}
-  def handle(%{"data" => data}) do
-    stripe_account = data |> retrieve_account
-    local_account = data |> load_account
-
-    local_account |> update(stripe_account)
-  end
-
-  defp retrieve_account(%{"object" => %{"id" => id_from_stripe}}) do
-    # hardcoded for testing
-    id_from_stripe = "acct_19JlxsFTVVt7Lv80"
-
-    {:ok, stripe_account} = @api.Account.retrieve(id_from_stripe)
-    stripe_account
-  end
-
-  defp load_account(%{"object" => %{"id" => id_from_stripe}}) do
-     # hardcoded for testing
-    id_from_stripe = "acct_19JlxsFTVVt7Lv80"
-
-    StripeConnectAccount
-    |> Repo.get_by(id_from_stripe: id_from_stripe)
-  end
-
-  defp update(%StripeConnectAccount{} = record, %Stripe.Account{} = stripe_account) do
-    {:ok, params} =
-      stripe_account
-      |> Adapters.StripeConnectAccount.to_params(%{})
-
-    record
-    |> StripeConnectAccount.webhook_update_changeset(params)
-    |> Repo.update
+  def handle(%{"data" => %{"object" => %{"id" => id_from_stripe}}}) do
+    with {:ok, %Stripe.Account{} = stripe_account} <-
+           @api.Account.retrieve(id_from_stripe),
+         %StripeConnectAccount{} = local_account <-
+           Repo.get_by(StripeConnectAccount, id_from_stripe: id_from_stripe),
+         {:ok, params} <-
+           stripe_account |> Adapters.StripeConnectAccount.to_params(%{})
+    do
+      local_account
+      |> StripeConnectAccount.webhook_update_changeset(params)
+      |> Repo.update
+    else
+      {:error, %Stripe.APIErrorResponse{}} -> {:error, :stripe_error}
```

Comment:
It could conceivable fail with this as well, so just a `nil` case is not enough. I also added a catch-all case, just in case (no pun intended).